process/peer/process.go

```diff
@@ -585,16 +687,12 @@ func (vs *validatorStatistics) updateValidatorInfo(validatorList []sharding.Vali
 		case leaderSuccess:
 			err = peerAcc.IncreaseLeaderSuccessRateWithJournal(1)
 			newRating = vs.rater.ComputeIncreaseProposer(peerAcc.GetTempRating())
-
 			if err != nil {
 				return err
 			}
 
 			leaderAccumulatedFees := core.GetPercentageOfValue(accumulatedFees, vs.rewardsHandler.LeaderPercentage())
 			err = peerAcc.AddToAccumulatedFees(leaderAccumulatedFees)
-		case leaderFail:
-			err = peerAcc.DecreaseLeaderSuccessRateWithJournal(1)
```

Comment:
Isn't this supposed to be kept here? When a leader does not propose a block for whatever reason, all peers should decrease the rate

Reply:
leaderFail is not possible for this function. This is called for proposed, signed blocks. missed blocks update is done in another function.

Reply:
in decreaseAll